lib/rubocop/cop/style/array_join.rb

```diff
@@ -18,6 +18,14 @@ def on_send(node)
 
           add_offense(node, :selector)
         end
+
+        private
+
+        def autocorrect(node)
+          expr = node.loc.expression
+          new_expr = expr.source.sub(/\s*\*\s*(.*$)/, '.join(\1)')
```

Comment:
Generally it's advisable to use node manipulation instead of regular expressions. With regexps there's also some potential for false positives in edge cases.